### PR TITLE
DROOLS-3772 DMN DT Analysis Rule masking

### PR DESCRIPTION
**Pull request by tarilabs**
@etirelli @baldimir @jiripetrlik

### CODE REVIEW

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/model/DTAnalysis.java [hunk 1]
```diff
@@ -194,4 +218,53 @@ private Collection gapsAsMessages() {
         return results;
     }
 
+    public void computeMaskedRules() {
+        if (sourceDT.getHitPolicy() != HitPolicy.PRIORITY) {
+            return;
+        }
+        for (Overlap overlap : overlaps) {
```
Comment (baldimir) [inline kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/model/DTAnalysis.java:238]: Could this be decomposed to smaller pieces somehow please?

**Review by baldimir — COMMENTED** [review]
Please see comment inline.

File: kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/model/DTAnalysis.java (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-validation/src/main/java/org/kie/dmn/validation/dtanalysis/model/DTAnalysis.java:238]: Proposal committed @baldimir

**Comment by tarilabs** [comment]
(rebased to resolve conflicts)

**Review by baldimir — APPROVED** [review]
Approving with a note that the code I commented should be probably split more. I will take a look at that later.

**Pull request merged**